game/unicode.cc

```diff
@@ -37,62 +39,123 @@ std::string UnicodeUtil::getCharset (std::string const& str) {
 	return MimeEncodingName(encoding);
 }
 
-void UnicodeUtil::convertToUTF8 (std::stringstream &_stream, std::string _filename) {
-	std::string data = _stream.str();
-	std::string charset = UnicodeUtil::getCharset(data);	
+std::string UnicodeUtil::convertToUTF8 (std::string const& str, std::string _filename, CaseMapping toCase, bool assumeUTF8) {
```

Comment:
```suggestion
std::string UnicodeUtil::convertToUTF8 (std::string const& str, const std::string &_filename, CaseMapping toCase, bool assumeUTF8) {
```

Reply:
actually filename seems really useless here... (ok code was already like this)

Reply:
Filename is only used for providing info to the user, in case they want to take the time of fixing the encodings.

Reply:
Ok, that was already like this, but this would deserve rework as well IMHO.